hypnos/src/main.c

```diff
@@ -45,58 +45,33 @@ void main(void)
 	event_handler_init();
 	gfx_update();
 	backlight_init();
-
-	while (true) {
-		if (bt_mode()) {
-			bt_thread();
-		} else {
-			main_thread();
-		}
-	}
 }
 
 /* ************* thread functions ***************/
 void main_thread(void)
 {
 	while (true) {
-	await_disable_bt:
 		bt_await_off();
+		LOG_INF("Disabling Bluetooth...");
 		bt_adv_stop();
-		while (true) {
-			k_sleep(SLEEP_TIME);
-			if (bt_mode()) {
-				gfx_bt_set_label(1);
-				gfx_update();
-				goto await_disable_bt;
-			}
-			k_cpu_idle();
-		}
+		cts_sync_enable(false);
+		k_cpu_idle();
 	}
 }
 
 void bt_thread(void)
 {
 	while (true) {
-	await_enable_bt:
 		bt_await_on();
-		LOG_INF("Entering bluetooth mode...");
+		LOG_INF("Enabling Bluetooth...");
 		if (bt_is_initialized()) {
 			bt_adv_start();
```

Comment:
I don't get what you want to do in those 2 thread.
Is that really starting advertisement/starting bluetooth on every run on the loop ?

Reply:
Nevermind, I just found the Bluetooth API from Zephyr.  😅

Reply:
The threads are needed so that the entire application doesn't crash when enabling/disabling advertising. This was something that couldn't be done in ISRs. I'm sure there are other ways to do this, but I'll keep it like this for now. By the way, I changed the log message to "Enabling BLE advertising..."